spectec/src/il/validation.ml

```diff
@@ -478,6 +470,11 @@ and valid_path env p t : typ =
   t'
 
 and valid_iterexp env (iter, ids) : env =
+  let iter =
```

Comment:
Ah, sorry for not being clearer. We still need to validate the variable. We only need to strip it for the purpose of the eq_iter check below. Because this combination should be legal:
```
(...x...)^n          (...x...)^(i<n)
```

Reply:
Oh, I see. I fixed it.
Thanks for the comments.

Reply:
Nit: can we name this iter' to avoid confusion?